cpp/kiss_icp/core/Registration.hpp

```diff
@@ -30,9 +30,19 @@
 
 namespace kiss_icp {
 
-Sophus::SE3d RegisterFrame(const std::vector<Eigen::Vector3d> &frame,
-                           const VoxelHashMap &voxel_map,
-                           const Sophus::SE3d &initial_guess,
-                           double max_correspondence_distance,
-                           double kernel);
+struct Registration {
+    explicit Registration(int max_num_iteration, double estimation_threshold)
+        : max_num_iterations_(max_num_iteration), estimation_threshold_(estimation_threshold) {}
+
+    // Register a point cloud to the given internal map representation. The config input parameter
+    // contains all the necessary parametrization for the ICP loop
+    Sophus::SE3d AlignCloudToMap(const std::vector<Eigen::Vector3d> &frame,
+                                 const VoxelHashMap &voxel_map,
+                                 const Sophus::SE3d &initial_guess,
+                                 double max_distance,
```

Comment:
I think I prefer the previous `max_correspondence_distance` to be more clear what kind of distance `AlignCloudToMap` expects here. Otherwise, could also be a max distance of map points we consider for registration for example.

Reply:
Agree 👍. What about the new method name? I didn't like the overuse of the word registration but couldn't come up with a better name 😂

Reply:
Maybe `AlignPointsToMap`? That aligns better with the function arguments.

But I am fine with both!

Reply:
done